pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -1151,7 +1151,10 @@ async def _responses_create(
 
         if not tools:
             tool_choice: Literal['none', 'required', 'auto'] | None = None
-        elif not model_request_parameters.allow_text_output:
+        elif (
+            not model_request_parameters.allow_text_output
+            and OpenAIModelProfile.from_profile(self.profile).openai_supports_tool_choice_required
```

Comment:
We have another call to `profile = OpenAIModelProfile.from_profile(self.profile)` further down; can you please move the variable to the top and use it in both places?